### PR TITLE
Feature/mat 6027 mongock patientid

### PR DESCRIPTION
**Pull request by nmorasb**
## MADiE PR

Jira Ticket: [MAT-6027](https://jira.cms.gov/browse/MAT-6027)
(Optional) Related Tickets:
[MAT-6002](https://jira.cms.gov/browse/MAT-6002)

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  public static String replacePatientRefs(String json, String newPatientId) {
+    final String pattern = "(?i)(\"reference\"\\s*:\\s*\"Patient\\/[A-Za-z0-9\\-\\.]{1,64}\")";
+    return json.replaceAll(pattern, "\"reference\": \"Patient/" + newPatientId + "\"");
+  }
+
+  public static String replaceFullUrlRefs(String json, String oldFullUrl, String newPatientId) {
+    final String pattern = "(\"reference\"\\s*:\\s*\"" + Pattern.quote(oldFullUrl) + "\")";
+    return json.replaceAll(pattern, "\"reference\": \"Patient/" + newPatientId + "\"");
```
Comment (RohitKandimalla) [inline src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java:92]: To my knowledge Full Urls are not a references to patient, or am I misunderstanding this piece of code.

File: src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  public static boolean isUuid(String value) {
+    try {
+      UUID.fromString(value);
+      return true;
+    } catch (Exception exception) {
+      // handle the case where string is not valid UUID
```
Comment (RohitKandimalla) [inline src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java:100]: Do we have to handle this ?

File: src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java (hunk 1 above)
Comment (RohitKandimalla) [reply src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java:92]: ooh there could be a references to full Url, Wasn't aware of that

**Review by RohitKandimalla — APPROVED** [review]

**Review by adongare — APPROVED** [review]

File: src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java (hunk 2 above)
Comment (nmorasb) [reply src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java:100]: no...lol bad comment - in this case we don't care, we just return false to indicate that it is not a UUID

File: src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java (hunk 1 above)
Comment (nmorasb) [reply src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java:92]: yeah exactly - a reference could contain a fullUrl, though I never saw any JSON doing that..but still wanted to handle that case before we whack all the old fullUrls like we did with the Patient IDs..

**Pull request merged**